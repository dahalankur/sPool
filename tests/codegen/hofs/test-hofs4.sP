def quack hi():
    println("HI")
    return;

def quack hof((quack -> quack) f):
    f()
    def quack nested():
        println("nested")
        f()
        return;
    nested()
    return;

hof(hi)

hof(lambda quack ():
    println("QUACK")
    return;)
