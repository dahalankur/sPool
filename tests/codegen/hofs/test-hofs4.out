HI
nested
HI
QUACK
nested
QUACK
